### PR TITLE
Add test for class redef callsite propagation

### PR DESCRIPTION
**Pull request by theresa-m**
First step for issue: #4542
Tests for: #5029

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by theresa-m** [comment]
fyi @gacholio @DanHeidinga these tests go along with #5402

File: test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc020.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+		/* Redefine class with more than one method with an indy call. */
+		if (!runTest("testIndyRedefinition (multi class test)", new rc020_testIndyAsmGenerator().generateMultiMethodClass(), 1, 0)) {
+		 	return false;
+		}
+
+		/* Redefine class with one indy call where bsm has three args with a double slot in the middle. If the static arguments
+		 * to the BSM change, the output will also change and BSM should be reinvoked. Verify that values are checked correctly.
+		 */
+		if (!runTest("testIndyRedefinition (basic test, bsm double slot with three args parameter change- should fail)",
```
Comment (theresa-m) [inline test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc020.java]: there is one test here that changes the static arguments for the same bsm where the callsite shouldn't be copied. I'll add one where the NAS changes as well for good measure.

File: runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc020.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+jboolean JNICALL
+Java_com_ibm_jvmti_tests_redefineClasses_rc020_redefineClass(JNIEnv * jni_env, jclass klass, jclass originalClass, jint classBytesSize, jbyteArray classBytes)
+{
+	JVMTI_ACCESS_FROM_AGENT(env);
+	jbyte * class_bytes;
+	char * classFileName = env->testArgs;
+	jvmtiClassDefinition classdef;
+	jvmtiError err;
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/redefineClasses/rc020.c]: updated.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,win,xlinux jdk8

**Comment by theresa-m** [comment]
xlinux testing failed due to "Cannot contact ub16x64j99: hudson.remoting.ChannelClosedException: Channel "unknown": Remote call on ub16x64j99 failed. The channel is closing down or has closed down"

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk8

**Pull request merged**